climakitae/explore/warming.py

```diff
@@ -129,13 +161,36 @@ def clean_list(data, gwl_times):
     return data.sel(all_sims=keep_list)
 
 
-def get_sliced_data(y, years, window=15, anom="Yes"):
+def clean_warm_data(warm_data):
+    """
+    Cleaning the warming levels data in 3 parts:
+      1. Removing simulations where this warming level is not crossed. (centered_year)
+      2. Removing timestamps at the end to account for leap years (time)
+      3. Removing simulations that go past 2100 for its warming level window (all_sims)
+    """
+    # Cleaning #1
+    warm_data = warm_data.sel(all_sims=~warm_data.centered_year.isnull())
+
+    # Cleaning #2
+    warm_data = warm_data.isel(
+        time=slice(0, len(warm_data.time) - 1)
+    )  # -1 is just a placeholder for 30 year window, this could be more specific.
+
+    # Cleaning #3
+    # warm_data = warm_data.dropna(dim="all_sims")
+
+    return warm_data
+
+
+def get_sliced_data(y, level, years, window=15, anom="Yes"):
```

Comment:
The window is really `30` years in the case of `15` so a little confusing naming. Could take the window and then halve it and +/- it to get the time range.

Reply:
I see that you are trying to avoid the fractional year window issue...

Reply:
This is a great point. I'm not too sure how to change this for now (what is half a window called?). I'm going to leave it as is because the documentation already listed provides enough detail describing what `window` is, and this method is only using in our back-end, so at least it shouldn't be confusing for users. However, I'm going to include this as another bullet in the `GWL Future Improvements` doc. Thanks for catching this.